gateway/ln-gateway/src/lib.rs

```diff
@@ -849,9 +848,27 @@ impl Gateway {
         .map_err(|e| GatewayError::UnexpectedState(e.to_string()))
     }
 
+    /// Requests the gateway to pay an outgoing LN invoice. Returns the payment
+    /// hash's preimage on success.
+    async fn handle_pay_normal_invoice_msg(&self, payload: PayInvoicePayload) -> Result<Preimage> {
+        if let GatewayState::Running { lightning_context } = self.get_state().await {
+            let res = lightning_context
+                .lnrpc
+                .pay(payload.invoice, payload.max_delay, payload.max_fee)
+                .await?;
+            Ok(Preimage(res.preimage.try_into().unwrap()))
+        } else {
+            warn!("Gateway is not connected to lightning node, cannot pay invoice");
+            Err(GatewayError::Disconnected)
+        }
+    }
```

Comment:
handle_self_pay_invoice_msg , distinguish between this and when it's paying on behalf of a fedimint client's contract

Reply:
Done.